### DISCUSSION TITLE
Want to Obtain Relative Coordinates of Images After Conducting SLAM

### DISCUSSION THREAD

**Discussion by dokodoko123** [stella-cv/stella_vslam#598] [Q&A]
I'm currently performing position estimation using run_image_slam on panoramic images. Is it possible to obtain the relative coordinates of those images? If you know how, I would appreciate it if you could share.

**Comment by Skwangles** [comment]
In terms of finding the coordinates of the images I have done a similar thing with video - my coordinate extraction takes place after SLAM maps it, so is not real-time which might not fit your use case. If you need realtime, you'll likely need to fork Stella and do stuff with the keyframe pose_cw attribute.

Stella VSLAM puts positioning in 'poses' as 4x4 rotation and translation matrixes (from my understanding).

If your use case is for mapping two images in an already mapped scene, then you can follow a similar process to the publishers (socket, pangolin, etc.) - which when SLAM is running gets the current frames. Just modify your system to compare the poses of the two images as they get fed in.

In terms of making positions relative, I'm sure there is some matrix math you can do between rot/trans matrixes to find their relative positions.

Extra info if it suits your use case better - non-realtime method:
The map output of stella, if put into sqlite3 mode 
_config.yaml:_ 
```
System:
  map_format: 'sqlite3' 
```
From my understanding, every image you pass in should become a 'keyframe' - Idk because my use case doesn't necessitate verifying that, I'll leave that up to you. 
The 'keyframes' are stored in a 'keyframes' table, in there is a field called pose_cw which provides the 'pose' or position and translation of the keyframe in the 3D space used for mapping. It is a 4x4 rotation and translation matrix, which if you have a look at the socket_publisher and socket_publisher code you can get an idea for how it can be decoded from the BLOB.
